src/Resources/ResourceManager/Implementation/Deployments/TestAzureDeploymentCmdlet.cs

```diff
@@ -12,38 +12,152 @@
+        [Parameter(ParameterSetName = ResourceGroupParameterSetWithParameterlessTemplateObject,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource group name.")]
+        [Parameter(ParameterSetName = ResourceGroupParameterSetWithParameterlessTemplateFile,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource group name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(ParameterSetName = TenantParameterSetWithTemplateObjectParameterObject,
+            Mandatory = true, HelpMessage = "Create deployment at tenant scope if specified.")]
+        [Parameter(ParameterSetName = TenantParameterSetWithTemplateObjectParameterFile,
+            Mandatory = true, HelpMessage = "Create deployment at tenant scope if specified.")]
+        [Parameter(ParameterSetName = TenantParameterSetWithTemplateFileParameterObject,
+            Mandatory = true, HelpMessage = "Create deployment at tenant scope if specified.")]
+        [Parameter(ParameterSetName = TenantParameterSetWithTemplateFileParameterFile,
+            Mandatory = true, HelpMessage = "Create deployment at tenant scope if specified.")]
+        [Parameter(ParameterSetName = TenantParameterSetWithParameterlessTemplateObject,
+            Mandatory = true, HelpMessage = "Create deployment at tenant scope if specified.")]
+        [Parameter(ParameterSetName = TenantParameterSetWithParameterlessTemplateFile,
+            Mandatory = true, HelpMessage = "Create deployment at tenant scope if specified.")]
+        public SwitchParameter Tenant { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The deployment mode.")]
+        public DeploymentMode Mode { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Rollback to the last successful deployment in the resource group, should not be present if -RollBackDeploymentName is used.")]
+        public SwitchParameter RollbackToLastDeployment { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Rollback to the successful deployment with the given name in the resource group, should not be used if -RollbackToLastDeployment is used.")]
+        public string RollBackDeploymentName { get; set; }
+
         public override void ExecuteCmdlet()
         {
+            if (string.IsNullOrEmpty(this.ResourceGroupName) && this.Mode == DeploymentMode.Complete)
+            {
+                WriteExceptionError(new ArgumentException(ProjectResources.InvalidDeploymentMode));
+            }
+
+            if (this.RollbackToLastDeployment && !string.IsNullOrEmpty(this.RollBackDeploymentName))
+            {
+                WriteExceptionError(new ArgumentException(ProjectResources.InvalidRollbackParameters));
+            }
```

Comment:
The backend will respond with a 400 in these situation right? Do we even need these?

Reply:
Back-compat maybe?

---
In reply to: [308505704](https://github.com/Azure/azure-powershell/pull/9707#discussion_r308505704) [](ancestors = 308505704)

Reply:
Right. it's for back-compat, making sure for resource group level it works the same as New-AzResourceGroupDeployment.

Reply:
Makes sense.

---
In reply to: [308965884](https://github.com/Azure/azure-powershell/pull/9707#discussion_r308965884) [](ancestors = 308965884)